main.go

```diff
@@ -183,21 +183,11 @@ func main() {
 }
 
 func setupControllerConfig(mgr ctrl.Manager) error {
-	operatorNamespace, err := infrastructure.GetOperatorNamespace()
-	if err == nil {
-		config.ConfigMapReference.Namespace = operatorNamespace
-	} else {
-		config.ConfigMapReference.Namespace = os.Getenv(infrastructure.WatchNamespaceEnvVar)
-	}
-	err = config.WatchControllerConfig(mgr)
```

Comment:
Che Operator needs to be adapted to these changes not to break Che + DevWorkspace integration on Kubernetes: 

https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/dev-workspace/dev_workspace.go#L332

please note that Operator CR must be already available when Che Operator does this logic

Reply:
Opened PR https://github.com/eclipse-che/che-operator/pull/1081 (note no config CRD is necessary to run DWO anymore, and `clusterHostSuffix` is only required for the basic routing solver, which is outside the Che Operator use-case).